src/presentation/ui/page/databases.templ

```diff
@@ -0,0 +1,352 @@
+					'POST',
+					'/api/v1/services/installables/',
+					{
+						swap: 'none',
+						values: { name: databaseType },
+					},
+				);
+			},
+			get shouldDisableCreateDatabaseSubmitButton() {
+				return this.database.name == ''
+			},
+			get shouldDisableCreateDatabaseUserSubmitButton() {
+				return this.database.name == '' || this.databaseUser.username == '' || this.databaseUser.password == ''
+			},
+			generateRandomUserPassword() {
+				const passwordLength = 16;
+				const chars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_+';
+
+				let passwordContent = '';
+				let passwordIterationCount = 0;
+				while (passwordIterationCount < passwordLength) {
+					const randomIndex = Math.floor(Math.random() * chars.length);
+					const indexAfterRandomIndex = randomIndex + 1;
+					passwordContent += chars.substring(randomIndex, indexAfterRandomIndex);
+
+					passwordIterationCount++;
+				}
+
+				this.databaseUser.password = passwordContent;
+			},
+
+			// Modal states
+			isCreateDatabaseModalOpen: false,
+			openCreateDatabaseModal() {
+				this.resetPrimaryStates();                                                                                                                                                                                                  
+
+				this.isCreateDatabaseModalOpen = true;
+			},
+			closeCreateDatabaseModal() {
+				this.isCreateDatabaseModalOpen = false;
+			},
+			isDeleteDatabaseModalOpen: false,
+			openDeleteDatabaseModal(databaseName) {
+				this.resetPrimaryStates();
+
+				this.database.name = databaseName;
+				this.isDeleteDatabaseModalOpen = true;
+			},
+			closeDeleteDatabaseModal() {
+				this.isDeleteDatabaseModalOpen = false;
+			},
+			deleteDatabaseElement(databaseType) {
+				htmx.ajax(
+					'DELETE',
+					'/api/v1/database/' + databaseType + '/' + this.database.name + '/',
+					{swap: 'none'},
+				);
+				this.closeDeleteDatabaseModal();
+			},
+			isCreateDatabaseUserModalOpen: false,
+			openCreateDatabaseUserModal() {
+				this.resetPrimaryStates();
+
+				this.isCreateDatabaseUserModalOpen = true;
+			},
+			closeCreateDatabaseUserModal() {
+				this.isCreateDatabaseUserModalOpen = false;
+			},
+			isDeleteDatabaseUserModalOpen: false,
+			openDeleteDatabaseUserModal(databaseName, databaseUsername) {
+				this.resetPrimaryStates();
+
+				this.database.name = databaseName;
+				this.databaseUser.username = databaseUsername;
+				this.isDeleteDatabaseUserModalOpen = true;
+			},
+			closeDeleteDatabaseUserModal() {
+				this.isDeleteDatabaseUserModalOpen = false;
+			},
+			deleteDatabaseUserElement(databaseType) {
+				htmx.ajax(
+					'DELETE',
+					'/api/v1/database/' + databaseType + '/' + this.database.name + '/user/' + this.databaseUser.username + '/',
+					{swap: 'none'},
+				);
+				this.closeDeleteDatabaseUserModal();
+			},
+		}))
+	})
+}
+
+templ DatabasesIndex(selectedDatabaseTypeOverview presenterDto.DatabaseOverview) {
```

Comment:
I think databaseType here is redundant. selectedDbOverview for instance wouldn't hurt legibility.

Reply:
Done!